state_test.go

```diff
@@ -2,12 +2,73 @@ package tfjson
 
 import (
 	"encoding/json"
+	"io/ioutil"
 	"os"
 	"testing"
 )
 
-func TestStateValidate(t *testing.T) {
-	f, err := os.Open("test-fixtures/no_changes/plan.json")
+func TestStateValidate_raw(t *testing.T) {
```

Comment:
Is this test useful if validate is being called anyway?

Reply:
It tests that validation is idempotent and calling it explicitly - which many consumers of this library probably do today - doesn't hurt. So from that perspective I think it's useful.

You could argue "why not have the same test for other structs" - and my reasoning here was just that this one seemed to pose more real risk (due to state being commonly represented in two different ways) so I was generally more careful about testing, but I'm more than happy to add these.

Reply:
:+1: sounds good - just wanted to check the rationale!